weblab/entities/views.py

```diff
@@ -400,6 +404,33 @@ class ProtocolEntityVersionListView(ProtocolEntityTypeMixin, VersionListView):
     pass
 
 
+class EntityArchiveView(EntityVisibilityMixin, SingleObjectMixin, View):
+    """
+    Download a version of a model
+    """
+    model = Entity
+
+    def get(self, request, *args, **kwargs):
+        entity = self.get_object()
+        ref = self.kwargs['sha']
+        commit = entity.repo.get_commit(ref)
+
+        if not commit:
+            raise Http404
+
+        zipfile_name = os.path.join(
+            get_valid_filename('%s_%s.zip' % (entity.name, commit.hexsha))
+        )
+
+        archive = entity.repo.archive(ref)
+
+        response = HttpResponse(content_type='application/zip')
+        response['Content-Disposition'] = 'attachment; filename=%s' % zipfile_name
+        response.write(archive.read())
```

Comment:
I imagine it may cause a problem if datasets get large. Though we're only using this method for model and protocol downloads, as experiments are stored differently (i.e. already as a zip, so can be streamed directly from disk).

Reply:
OK, that'll be fine then! Models & protocols shouldn't have large files in them.